app/src/main/java/com/asma/tasky/feature_management/presentation/agenda/AgendaScreen.kt

```diff
@@ -1,11 +1,153 @@
 package com.asma.tasky.feature_management.presentation.agenda
 
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
 import androidx.compose.material.Text
-import androidx.compose.runtime.Composable
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowDropDown
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.asma.tasky.R
+import com.asma.tasky.core.domain.util.Util
+import com.asma.tasky.core.presentation.ui.theme.LightGray
+import com.asma.tasky.core.presentation.ui.theme.SpaceLarge
+import com.asma.tasky.core.presentation.ui.theme.SpaceMedium
+import com.asma.tasky.core.presentation.ui.theme.SpaceSmall
+import com.asma.tasky.feature_management.domain.AgendaItem
+import com.asma.tasky.feature_management.presentation.agenda.components.AgendaListItem
+import com.asma.tasky.feature_management.presentation.agenda.components.DayItem
+import com.asma.tasky.feature_management.presentation.agenda.components.Needle
+import java.time.LocalDate
+import java.util.*
 
 @Preview
 @Composable
-fun AgendaScreen() {
-    Text(text = "Agenda Screen")
+fun AgendaScreen(
+    viewModel: AgendaViewModel = hiltViewModel(),
+    onClick: (AgendaItem) -> Unit = {}
+) {
+    val state by viewModel.agendaState.collectAsState()
+
+    val pastItems by remember {
+        derivedStateOf {
+            state.items.filter {
+                it.startDate < System.currentTimeMillis() / 1000
+            }
+        }
+    }
```

Comment:
`derivedStateOf` is rather used for state that frequently changes which is not the case for agenda items. Here, you can simply use remember with keys:
```
val pastItems = remember(state.items) {
    state.items.filter { ... }
}
```